From libc-ports-return-536-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Fri Oct 27 00:44:39 2006
Return-Path: <libc-ports-return-536-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 9692 invoked by alias); 27 Oct 2006 00:44:39 -0000
Received: (qmail 9673 invoked by uid 22791); 27 Oct 2006 00:44:38 -0000
X-Spam-Status: No, hits=-1.2 required=5.0 	tests=AWL,BAYES_20,RCVD_ILLEGAL_IP,SPF_PASS,TW_CV,TW_DM
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.31) with ESMTP; Fri, 27 Oct 2006 00:44:36 +0000
Received: (qmail 23961 invoked from network); 27 Oct 2006 00:44:34 -0000
Received: from unknown (HELO digraph.polyomino.org.uk) (joseph@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 27 Oct 2006 00:44:34 -0000
Received: from jsm28 (helo=localhost) 	by digraph.polyomino.org.uk with local-esmtp (Exim 4.63) 	(envelope-from <joseph@codesourcery.com>) 	id 1GdFpp-00005M-9U 	for libc-ports@sourceware.org; Fri, 27 Oct 2006 00:44:33 +0000
Date: Fri, 27 Oct 2006 00:44:33 +0000 (UTC)
From: "Joseph S dot  Myers" <joseph at codesourcery dot com>
X-X-Sender: jsm28@digraph.polyomino.org.uk
To: libc-ports at sourceware dot org
Subject: Fix ARM recvfrom and sendto cancellation
Message-ID: <Pine.LNX.4.64.0610270042200.28434@digraph.polyomino.org.uk>
MIME-Version: 1.0
Content-Type: TEXT/PLAIN; charset=US-ASCII
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

The unwind information for 6-argument ARM EABI syscalls is incorrect, 
leading to failures of tst-cancelx4 and tst-cancelx5 relating to recvfrom 
and sendto.  This patch fixes those test failures.

2006-10-26  Joseph S. Myers  <joseph@codesourcery.com>

	* sysdeps/unix/sysv/linux/arm/eabi/nptl/sysdep-cancel.h
	(DOCARGS_6): Change ".pad #20" to ".pad #16".

---
 sysdeps/unix/sysv/linux/arm/eabi/nptl/sysdep-cancel.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: sysdeps/unix/sysv/linux/arm/eabi/nptl/sysdep-cancel.h
===================================================================
--- sysdeps/unix/sysv/linux/arm/eabi/nptl/sysdep-cancel.h.orig
+++ sysdeps/unix/sysv/linux/arm/eabi/nptl/sysdep-cancel.h
@@ -94,7 +94,7 @@
 # define UNDOCARGS_5	ldmfd sp!, {r0, r1, r2, r3}; .fnend; .fnstart; .save {r4}; .save {r7, lr}; .pad #4
 # define RESTORE_LR_5	ldmfd sp!, {r4, r7, lr}
 
-# define DOCARGS_6	.save {r4, r5}; stmfd sp!, {r0, r1, r2, r3, r7, lr}; .save {r7, lr}; .pad #20
+# define DOCARGS_6	.save {r4, r5}; stmfd sp!, {r0, r1, r2, r3, r7, lr}; .save {r7, lr}; .pad #16
 # define UNDOCARGS_6	ldmfd sp!, {r0, r1, r2, r3}; .fnend; .fnstart; .save {r4, r5}; .save {r7, lr}
 # define RESTORE_LR_6	RESTORE_LR_0
 
